### PR TITLE
Add Python API reference docs

### PR DESCRIPTION
**Pull request by adisbladis**

### CODE REVIEW

File: ci/lint-docs.sh [hunk 1]
```diff
@@ -3,6 +3,6 @@
 
 set -eux
 
-git ls-files | xargs codespell
+git ls-files | xargs codespell -L keypair,iam,hda
```
Comment (nlewo) [inline ci/lint-docs.sh:6]: Is `keypair` keyword still needed here since there are fixed?

File: ci/lint-docs.sh [hunk 2]
```diff
@@ -3,6 +3,6 @@
...
 sphinx-build -M clean doc/ doc/_build
-sphinx-build -nW doc/ doc/_build
+sphinx-build -n doc/ doc/_build
```
Comment (nlewo) [inline ci/lint-docs.sh:8]: Do we really need to remove `W`?

File: ci/lint-docs.sh (hunk 1 above)
Comment (adisbladis) [reply ci/lint-docs.sh:6]: Yes, the nix expressions still says `keyPair`.

File: ci/lint-docs.sh (hunk 2 above)
Comment (adisbladis) [reply ci/lint-docs.sh:8]: Yes :/
We have circular imports guarded by `typing.TYPE_CHECKING`. That's not a problem for Mypy but it is a problem for Sphinx.

Comment (nlewo) [reply ci/lint-docs.sh:8]: Ok :(

**Pull request merged**